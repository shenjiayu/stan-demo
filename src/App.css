/* CSS for the theader */
.app-header-container {
  height: 20px;
  padding: 20px;
  color: white;
  background: linear-gradient(to bottom, #7abcff 0%,#60abf8 44%,#2088f0 100%);
}

.app-header-left-title {
  position: relative;
  float: left;
  padding-left: 100px;
  font-size: 1.5em;
  line-height: 25px;
}

.app-header-right-container {
  position: relative;
  float: right;
  padding-right: 100px;
  line-height: 25px;
}

.app-header-login {
  color: white;
  text-decoration: none;
  margin-right: 20px;
}

.app-header-free-trial {
  color: white;
  background: linear-gradient(to top, #45484d 0%,#000000 100%);
  border: none;
  padding: 5px 10px;
  font-size: 1em;
}

/* Mobile and Tablet device support for the header */
@media only screen and (max-width: 768px) {
  .app-header-container {
    text-align: center;
    height: 50px;
  }

  .app-header-left-title, .app-header-right-container {
    padding: 0 0 10px 0;
    width: 100%;
  }
}

/* CSS for the content */
.app-content-container {
  flex: 1 0 auto;
  padding: 20px 120px 20px 120px;
}

.app-content-inner-container {
  text-align: center;
}

.app-content-header {
  height: 20px;
  padding: 20px;
  background: linear-gradient(to top, #45484d 0%,#000000 100%);
}

.app-content-card {
  position: relative;
  display: inline-block;
  cursor: pointer;
  width: 150px;
  margin: 0 10px;
}

.app-content-text {
  color: white;
  line-height: 25px;
  padding-left: 100px;
  font-size: 1.3em;
}

.app-content-text-center {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
}

.app-content-card-description {
  display: block;
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.app-content-placeholder {
  background-color: black;
  height: 200px;
  width: 100%;
}

/* Mobile and Tablet device support for the content */
@media only screen and (max-width: 768px) {
  .app-content-container {
    text-align: center;
    padding: 20px 20px;
  }
}

/* CSS for the footer */
.app-footer-container {
  flex-shrink: 0;
  right: 0;
  bottom: 0;
  left: 0;
  padding: 40px 120px 40px 120px;
  background: linear-gradient(to top, #45484d 0%,#000000 100%);
}

.app-footer-info {
  color: white;
}

.app-footer-social-logo-container {
  position: relative;
  float: left;
  padding: 20px 0 0 0;
}

.app-footer-social-store-container {
  position: relative;
  float: right;
  padding: 20px 0 0 0;
}

.app-footer-social-logo, .app-footer-social-store {
  margin-right: 20px;
}

/* Mobile and Tablet device support for the content */
@media only screen and (max-width: 768px) {
  .app-footer-container {
    text-align: center;
    padding: 20px 20px;
  }
  
  .app-footer-social-logo-container {
    padding: 10px;
    width: 100%;
  }
}