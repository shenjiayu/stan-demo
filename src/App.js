import React, { Component } from 'react';
import {
  Route
} from 'react-router-dom'
import './App.css';

// Import components
import Header from './components/header';
import Footer from './components/footer';

// Import screens
import HomeScreen from './screens/home-screen';
import SeriesScreen from './screens/series-screen';
import MoviesScreen from './screens/movies-screen';

class App extends Component {
  render() {
    return [
        <Header key="Header" />,
        <Route key="HomeScreen" exact path="/" component={HomeScreen} />,
        <Route key="SeriesScreen" path="/series" component={SeriesScreen} />,
        <Route key="MoviesScreen" path="/movies" component={MoviesScreen} />,
        <Footer key="Footer" />
    ];
  }
}

export default App;
