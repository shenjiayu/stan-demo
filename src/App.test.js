import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  BrowserRouter as Router
} from 'react-router-dom';

// Import components
import Header from './components/header';
import ContentHeader from './components/content-header';
import Footer from './components/footer';

// Import screens
import App from './App';
import SeriesScreen from './screens/series-screen';

configure({ adapter: new Adapter() });

it('renders the header and compares with snapshots without crashing', () => {
  const header = renderer.create(<Header />);
  let tree = header.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the content header and compares with snapshots without crashing', () => {
  const contentHeader = renderer.create(<ContentHeader />);
  let tree = contentHeader.toJSON();
  expect(tree).toMatchSnapshot();
});

it('asserts the content of the content header', () => {
  const contentHeader = shallow(<ContentHeader pageName="Series" />);
  expect(contentHeader.find('.app-content-text').text()).toEqual('Popular Series');
});

it('renders the footer and compares with snapshots without crashing', () => {
  const footer = renderer.create(<Footer />);
  let tree = footer.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the home screen without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
      <App />
    </Router>
    , div);
});

it('asserts the number of data retrieved from the server', async () => {
  await fetch('https://raw.githubusercontent.com/StreamCo/react-coding-challenge/master/feed/sample.json')
          .then(resp => resp.json())
          .then(data => {
            expect(data.entries.length).toEqual(100);
          });
});