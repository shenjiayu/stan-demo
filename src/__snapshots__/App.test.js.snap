// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the content header and compares with snapshots without crashing 1`] = `
<div
  className="app-content-header"
>
  <div
    className="app-content-text"
  >
    Popular 
  </div>
</div>
`;

exports[`renders the footer and compares with snapshots without crashing 1`] = `
<footer
  className="app-footer-container"
>
  <p
    className="app-footer-info"
  >
    Home  |  Terms and Conditions  |  Privacy Policy  |  Collection Statement  |  Help  |  Manage Account
  </p>
  <small
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Copyright © 2016 DEMO Streaming. All Rights Reserved.
  </small>
  <div
    className="app-footer-social-container"
  >
    <div
      className="app-footer-social-logo-container"
    >
      <img
        alt="facebook"
        className="app-footer-social-logo"
        height="30"
        src="facebook-white.svg"
      />
      <img
        alt="twitter"
        className="app-footer-social-logo"
        height="30"
        src="twitter-white.svg"
      />
      <img
        alt="instagram"
        className="app-footer-social-logo"
        height="30"
        src="instagram-white.svg"
      />
    </div>
    <div
      className="app-footer-social-store-container"
    >
      <img
        alt="appstore"
        className="app-footer-social-store"
        height="30"
        src="app-store.svg"
      />
      <img
        alt="playstore"
        className="app-footer-social-store"
        height="30"
        src="play-store.svg"
      />
      <img
        alt="windowsstore"
        className="app-footer-social-store"
        height="30"
        src="windows-store.svg"
      />
    </div>
  </div>
</footer>
`;

exports[`renders the header and compares with snapshots without crashing 1`] = `
<header
  className="app-header-container"
>
  <div
    className="app-header-left-title"
  >
    DEMO Streaming
  </div>
  <div
    className="app-header-right-container"
  >
    <a
      className="app-header-login"
      href=""
    >
      Log in
    </a>
    <button
      className="app-header-free-trial"
    >
      Start your free trial
    </button>
  </div>
</header>
`;
